nshutijonathan/Auto-Mart
# Challenge2 feedbacks

#### What does this PR do?
Implemented challenge2 feedback
#### Description of Task to be completed?
 - provided buyer token instead of req.body.buyer
- Used object destructor post methods in API
- Fixed some response bugs 
- Added more tests
Have The following endpoint working in POSTMAN
*  POST/localhost:3000/api/v2/auth/signup
*  POST/localhost:3000/api/v2/auth/signin
*  POST/localhost:3000/api/v2/auth/signup/admin
*  GET/localhost:3000/api/v2/users
*  GET/localhost:3000/api/v2/users/<:id>
*   DELETE/localhost:3000/api/v2/users/<:id>
*   POST/localhost:3000/api/v2/car
*   GET/localhost:3000/api/v2/cars
#### How should this be manually tested?
* Download or clone the tutorial project code from [this repo](https://github.com/nshutijonathan/Auto-Mart), Install all required NPM  packages by running NPM install from the command line in the project root folder (where the package.json is located).

* Start the API by running NPM  start from the command line in the project root folder, you should see the message Server listening on port 3000. You can test the API directly using an application such as Postman or you can test it with one of the single pages in the browser.
* You must have node environment configured  and .env file in the root folder
* to test authentication and authorization, add this to the header 
Key: :x-auth-token,value:JWT TOKEN
#### Any background context you want to provide?
* There is Implementation of Javascript(ES6) logic with s data structures  to persist some data
